File: 804-unique-morse-code-words/804-unique-morse-code-words.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
                vector<string>code{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        set<string>out;
        for(auto s:words)
        {
            string cur="";
            for(char c:s)
            {
                cur+=code[c-'a'];
            }
            out.insert(cur);
        }
        return out.size();
    }
};

File: 804-unique-morse-code-words/804-unique-morse-code-words.cs
public class Solution {
    public int UniqueMorseRepresentations(string[] words) {
        HashSet<String>temp = new HashSet<String>();
        string [] k={".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        foreach(string s in words)
        {
            string cur="";
            foreach(char c in s)
            {
                cur+=k[(int)c-97];
            }
            temp.Add(cur);
        }
        return temp.Count;
    }
}